src/components/PadBtn/PadBtn.tsx

```diff
@@ -17,43 +35,143 @@ const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
     }
 
     if (input === "=") {
+      if (newCalc.result === 0) newCalc.result = parseFloat(mainStr);
+      setPastEquations(
+        pastEquations.concat(
+          <div key={pastEquations.length} className="history">
+            <p>{mainStr}</p>
+            <p>= {newCalc.result}</p>
+          </div>
+        )
+      );
+
+      if (newCalc.result !== 0) mainStr = newCalc.result.toString();
+      newCalc.result = 0;
+      newCalc.displayedString = mainStr;
+      setCalc(newCalc);
       return;
     }
 
     if (input === "<-") {
-      newCalc.displayedString = newCalc.displayedString.slice(0, -1);
+      mainStr = mainStr.slice(0, -1);
+      if (mainStr === "") mainStr = "0";
+      newCalc.displayedString = mainStr;
       getCalculation(newCalc);
       setCalc(newCalc);
       return;
     }
 
-    if (newCalc.displayedString.charAt(0) === "0")
-      newCalc.displayedString = newCalc.displayedString.slice(0, -1); //clear first zero
+    if (input === ",") input = ".";
 
-      let isLessThan8 = newCalc.nums[newCalc.nums.length - 1].length < 8;
-      if (isLessThan8) newCalc.displayedString += input; //need improvement
+    if (input === "xⁿ") input = "^";
 
+    if (input === "±" && mainStr === "0") return;
+
+    let lastDigit = mainStr.charAt(mainStr.length - 1);
+
+    if (!isNumber(lastDigit) && !isNumber(input))
+      //if (input !== "π" && input !== "√" && lastDigit !== "π" && lastDigit !== "√")
+      mainStr = mainStr.slice(0, -1);
+
+    if (mainStr.slice(-1) === "0" && isNumber(input))
+      mainStr = mainStr.slice(0, -1); //clear first zero
+
+    let numberLenght = 0;
+    for (let i = 0; i < mainStr.length; i++) {
+      if (isNumber(mainStr.charAt(i))) numberLenght++;
+      else numberLenght = 0;
+    }
+
+    if (numberLenght < 8 || !isNumber(input)) mainStr += input;
+
+    if (input === "±") {
+      let numberToInvert = newCalc.nums[newCalc.nums.length - 1];
+      mainStr = mainStr.slice(0, -1);
+      if (newCalc.sign[newCalc.sign.length - 1] === "-") {
```

Comment:
[suggestion] You could think of a way to make this more readable, maybe you could create variables and name them in a way that describes what you're doing each step of the way, so instead of just `mainStr` you would have a const for each process you're doing with the string and, if you need, you can just assign the resulting value to `mainStr` for last.

Reply:
Alternatively, you could just create variables for your regular expressions with a descriptive name, that should be enough to make your code more understandable.

Reply:
You can do this process everywhere you're using regular expressions.